### PR TITLE
[MM-37351] Split settings modal

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR splits settings modal into two

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-37351

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Screenshots

https://user-images.githubusercontent.com/2340127/129442749-7f10eb22-a533-4e97-8cde-450c4414cee0.mp4

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/user_settings/modal/index.ts [hunk 1]
```diff
@@ -31,6 +36,8 @@ function mapStateToProps(state: GlobalState) {
         sendEmailNotifications,
         requireEmailVerification,
         collapsedThreads,
+        globalHeaderEnabled: getGlobalHeaderEnabled(state),
+        isContentChannelPreferences: ownProps.isContentChannelPreferences,
```
Comment (devinbinnie) [inline components/user_settings/modal/index.ts:40]: This line is unnecessary, it should pass this through anyways.

**Review by devinbinnie — APPROVED** [review]
One non-blocking comment, but otherwise LGTM

File: components/user_settings/modal/user_settings_modal.tsx [hunk 2]
```diff
@@ -326,10 +339,17 @@ class UserSettingsModal extends React.PureComponent<Props, State> {
...
-                        />
+                        {this.props.globalHeaderEnabled && this.props.isContentChannelPreferences ? (
+                            <FormattedMessage
+                                id='channel_header.channelPreferences'
+                                defaultMessage='Channel Preferences'
+                            />
+                        ) : (
+                            <FormattedMessage
+                                id='user.settings.modal.title'
+                                defaultMessage='Account Settings'
```
Comment (nevyangelova) [inline components/user_settings/modal/user_settings_modal.tsx:350]: nit: maybe we should keep the naming consistent, account or profile.

File: components/user_settings/modal/index.ts (hunk 1 above)
Comment (nevyangelova) [reply components/user_settings/modal/index.ts:40]: why do we need this check? It seems like the globalHeaderEnabled could be enough for the use case. Also we don't need to add it to the index.ts if its coming from ownProps

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: components/user_settings/modal/index.ts (hunk 1 above)
Comment (iomodo) [reply components/user_settings/modal/index.ts:40]: @nevyangelova, @devinbinnie agree on the `ownProps` 

@nevyangelova When `globalHeaderEnabled` is on we can have 2 modals `Channel Preferences` or `Account Settings`. `isContentChannelPreferences`  flag determines which one to show.

File: components/user_settings/modal/user_settings_modal.tsx (hunk 2 above)
Comment (iomodo) [reply components/user_settings/modal/user_settings_modal.tsx:350]: `Account Settings` is the Title of the modal. `Profile` is the tab. So was in the figma files.

**Comment by esethna** [comment]
@iomodo confirming that accessing the "Account Settings" works when in Boards or Playbooks?

**Review by esethna — APPROVED** [review]
Approving on behalf of UX. QA please confirm the Account Settings are accessible from Boards and Playbooks

**Review by nevyangelova — APPROVED** [review]
LGTM :)

**Comment by jespino** [comment]
@iomodo @esethna Not necessarily for this PR, but the theme is for the whole suite, not only Channels, so maybe that should be in the user settings.

**Comment by jgilliam17** [comment]
Thanks @iomodo 
Tested, looks good to merge.
- Verified Old Account Settings have been split into two new settings modals and all existing settings options on both modals are working as before
**_Channel Preference_** --  accessible from global header gear icon; opens with Messaging specific options Notifications, Display, Sidebar and Advanced
**_Account Settings_** -- accessible from global header profile icon; opens with options Profile and Security
_**Boards and Playbooks**_ > QA will test both new settings modals on daily after merge. Currently test servers are not allowing manual plugin upload - not able to verify on the PR server.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**